### PR TITLE
[ENGG-1417] chore: rules table refractoring V2

### PR DESCRIPTION
**Pull request by wrongsahil**
Closes issue: 

## ✅ Checklist:

- [ ] Make sure linting and unit tests pass.
- [ ] No install/build warnings introduced.
- [ ] Verified UI in browser.
- [ ] For UI changes, added/updated analytics events (if applicable).
- [ ] For changes in extension's code, manually tested in Chrome and Firefox.
- [ ] For changes in extension's code, both MV2 and MV3 are covered.
- [ ] Added/updated unit tests for this change.
- [ ] Raised pull request to update corresponding documentation (if already exists).

### CODE REVIEW

**Review by nsrCodes — APPROVED** [review]
Looks good to me, just need to add some clarifications to the guidelines doc about the decisions taken here

File: app/src/componentsV2/ContentList/components/ContentListTable/components/BulkActionBar/bulkActionBar.scss [hunk 1]
```diff
@@ -1,4 +1,4 @@
-.bulk-action-bar-container {
+.rq-bulk-action-bar-container {
```
[the remark this answers is not in the document]
Comment (wrongsahil) [reply app/src/componentsV2/ContentList/components/ContentListTable/components/BulkActionBar/bulkActionBar.scss:1]: Done

File: app/src/componentsV2/SecondarySidebarLink/index.tsx [hunk 2]
```diff
@@ -0,0 +1,25 @@
+import React from "react";
+import { NavLink } from "react-router-dom";
+import "./SecondarySidebarLink.css";
+
+interface Props<T = string> {
```
[the remark this answers is not in the document]
Comment (wrongsahil) [reply app/src/componentsV2/SecondarySidebarLink/index.tsx]: Fixed

File: app/src/componentsV2/SecondarySidebarLink/index.tsx
[the remark this answers is not in the document]
Comment (wrongsahil) [reply app/src/componentsV2/SecondarySidebarLink/index.tsx:1]: No as items in the SecondarySidebar depends on the feature. 

Created a common component `componentsV2/SecondarySidebar`

File: app/src/features/rules/modals/RenameGroupModalWrapper/index.tsx [hunk 3]
```diff
@@ -0,0 +1,27 @@
...
+  useEffect(() => {
+    const openModal = (group: Group) => {
+      setGroupToRename(group);
+      setModalActive(true);
+    };
+
+    setOpenRenameGroupModal(() => openModal);
+  }, [setOpenRenameGroupModal]);
+
+  const onClose = () => {
```
[the remark this answers is not in the document]
Comment (wrongsahil) [reply app/src/features/rules/modals/RenameGroupModalWrapper/index.tsx:21]: No. In almost all the cases modal only controls when to close it as Modal is a blocking activity.
**Cases**
- Clicking on X
- Clicking outside Modal
- Modal completed some operation


When we need it in future, we can expose it outside too

File: app/src/features/rules/types/rules.ts [hunk 4]
```diff
@@ -275,8 +275,6 @@ export type Rule =
   | DelayRule
   | ScriptRule;
 
-export interface Group extends GroupRecord {
-  children: Rule[]; // aren't always correctly populated, but present
-}
+export interface Group extends GroupRecord {}
```
[the remark this answers is not in the document]
Comment (wrongsahil) [reply app/src/features/rules/types/rules.ts:278]: Nope not yet. Will fix it separately

File: app/src/features/rules/components/RulesList/components/RulesListIndex/components/index.ts
[the remark this answers is not in the document]
Comment (wrongsahil) [reply app/src/features/rules/components/RulesList/components/RulesListIndex/components/index.ts:1]: Done
https://www.notion.so/requestly/WebApp-Guidelines-8148ba612da040e496c4e2c86108f103?pvs=4#d541e5b0a74a4dfe9f77a3220134bd75

File: app/src/features/rules/context/modals.tsx [hunk 5]
```diff
@@ -0,0 +1,72 @@
+/**
+ * Contains all the common modals and state
+ */
+
+import React, { createContext, useContext, useState } from "react";
+import { Group, Rule, StorageRecord } from "../types/rules";
+
+type RulesModalsContextType = {
+  openCreateGroupModal: () => void;
+  setOpenCreateGroupModal: React.Dispatch<React.SetStateAction<() => void>>;
```
[the remark this answers is not in the document]
Comment (wrongsahil) [reply app/src/features/rules/context/modals.tsx]: Fixed. Added action suffix

**Pull request merged**